plugins/module_utils/common.py

```diff
@@ -178,6 +178,13 @@ def _raise_or_fail(exc, msg):
             if env_value is not None:
                 if AUTH_ARG_SPEC[arg_name].get("type") == "bool":
                     env_value = env_value.lower() not in ["0", "false", "no"]
+                if true_name == 'no_proxy' and LooseVersion(kubernetes.__version__) < LooseVersion("19.15.0"):
```

Comment:
Could you move this down to line 221, after the `if auth_set("host")` block? It's currently only checking if NO_PROXY has been set through the environment variable. It needs to check against the `auth` dict after we're done creating it. So something like:
```
if auth_set("no_proxy") and LooseVersion...
```